### PR TITLE
Newsletters: add 130 (2021-01-06)

### PR DESCRIPTION
**Pull request by harding**
Happy 2021!

- [ ] @harding update releases/RCs on Tuesday; ensure 0.21.0rc4 is actually uploaded
- [x] @bitschmidty `Bitcoin Core #19137` and `Bitcoin Core #20365`
- [x] @jnewbery `Bitcoin Core #20599`
- [x] @jonatack `Bitcoin Core GUI #162`
- [x] @dongcarl `Rust Bitcoin #499` and possibly related PRs.  This will be our first mention of Rust Bitcoin!

### CODE REVIEW

File: _posts/en/newsletters/2021-01-06-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+## News
+
+- **Proposed updates to generic signmessage:** Andrew Poelstra
+  [posted][poelstra post] a partial rewrite of the [BIP322][]
+  generic message signing proposal.  The rewrite mainly clarifies the
+  signing and verification procedures, but it now also allows wallets
+  that don't implement a complete set of checks to return an
+  "inconclusive" state for signatures that use scripts they don't
+  understand.  For example, a developer who wants to quickly implement
+  generic signmessage can only implement signature verification for
```
Comment (jonatack) [inline _posts/en/newsletters/2021-01-06-newsletter.md:25]: maybe s/can/might|might choose to/ 

(to clarify that its a choice by the developer)

File: _posts/en/newsletters/2021-01-06-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  more comprehensive verification tool.
+
+- **Trampoline routing:** Bastien Teinturier [posted][teinturier post]
+  to the Lightning-Dev mailing list a new [specification][tramp spec]
+  for [trampoline routing][topic trampoline payments] which allows a
+  spender to pay a receiver without calculating a route between their
+  nodes.  Instead, the spender routes their money to a nearby trampoline
+  node who calculates the next part of the route---getting the money
+  either to the receiver or to the next trampoline node.  This is useful
+  for spenders with lightweight LN clients that are often offline or
```
Comment (jonatack) [inline _posts/en/newsletters/2021-01-06-newsletter.md:38]: maybe use "that" for referring to both "node" and "clients"

File: _posts/en/newsletters/2021-01-06-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+- [Bitcoin Core 0.21.0rc4][Bitcoin Core 0.21.0] is a release candidate
+  for the next major version of this full node implementation and its
+  associated wallet and other software.  FIXME:harding ensure it gets uploaded to bitcoincore.org/bin before publication
+
+- [LND 0.12.0-beta.rc3][LND 0.12.0-beta] is the latest release candidate
+  for the next major version of this LN node.  It makes [anchor
+  outputs][topic anchor outputs] the default for commitment transactions
+  and adds support for them in its [watchtower][topic watchtowers]
+  implementation, reducing costs and increasing safety, and adds generic
```
Comment (jonatack) [inline _posts/en/newsletters/2021-01-06-newsletter.md:78]: s/and adds/and it adds/

File: _posts/en/newsletters/2021-01-06-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [Eclair 0.5.0][] is a new major version that adds support for a
+  scalable cluster mode (see [Newsletter #128][news128 akka]), block
+  chain watchdogs ([#123][news123 watchdog]), and additional plugin
+  hooks, among many other features and bug fixes.
+
+- [Bitcoin Core 0.21.0rc4][Bitcoin Core 0.21.0] is a release candidate
```
Comment (jonatack) [inline _posts/en/newsletters/2021-01-06-newsletter.md:70]: Is this list usually in alphabetical order? (e.g. Bitcoin Core before Eclair), ignore me if not

File: _posts/en/newsletters/2021-01-06-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+- [Bitcoin Core #20599][] net processing: Tolerate sendheaders and sendcmpct messages before verack FIXME:jnewbery
+
+- [Bitcoin Core #18772][] updates the `getblock` RPC when used with the
+  `verbosity=2` parameter to return a new `fee` field containing the
+  total fee for each transaction in the block.  This depends on
+  additional block undo data that nodes store in order to handle block
+  chain reorganizations.  Undo data is stored separately from other
+  block transaction data and nodes with pruning enabled may delete undo
+  data before deleting other date, so it's possible for pruned nodes to
```
Comment (jonatack) [inline _posts/en/newsletters/2021-01-06-newsletter.md:110]: s/date/data/

**Review by jonatack — COMMENTED** [review]
First light read-through. Shaping up to be a good newsletter to start the year!

File: _posts/en/newsletters/2021-01-06-newsletter.md (hunk 4 above)
Comment (harding) [reply _posts/en/newsletters/2021-01-06-newsletter.md:70]: Primary order is releases before RCs, which is why the Eclair release sorts before the Bitcoin Core RC.  Secondary order is the order of projects in the intro paragraph to the *Notable changes* section (which is my opinion on order of importance by type of project (full nodes > LN nodes > libraries > tools > documentation) and alphabetical within project type).

Comment (jonatack) [reply _posts/en/newsletters/2021-01-06-newsletter.md:70]: Thanks! Noting, since I've probably asked this before.

**Comment by jonatack** [comment]
@harding Not too sure how to describe the GUI Peers window (some people call it the tab window) and the peer details area/window. Opted to not set off header names with quotes or backticks for sobriety/not clutter things up; feel free to change.

**Comment by jnewbery** [comment]
Added 20599. Great newsletter so far! Thanks @harding and @jonatack.

**Comment by harding** [comment]
Pushed edits (thanks @jonatack and @jnewbery!) and reviewed the merge summaries (thanks @jonatack, @jnewbery, @dongcarl, and @bitschmidty!), with just a few small edits.

**Review by bitschmidty — COMMENTED** [review]
ACK 80ca2682aec94124a2b77eaf75f3b2cf79ac9f05

File: _posts/en/newsletters/2021-01-06-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+  `verack`. This PR restores the original behavior.
+
+- [Bitcoin Core #18772][] updates the `getblock` RPC when used with the
+  `verbosity=2` parameter to return a new `fee` field containing the
+  total fee for each transaction in the block.  This depends on
+  additional block undo data that nodes store in order to handle block
+  chain reorganizations.  Undo data is stored separately from other
+  block transaction data and nodes with pruning enabled may delete undo
+  data before deleting other data, so it's possible for pruned nodes to
+  sometimes return results without a `fee` field.
```
Comment (adamjonas) [inline _posts/en/newsletters/2021-01-06-newsletter.md:127]: nit: split infinitive
```suggestion
  data before deleting other data, so it's possible for pruned nodes
  sometimes to return results without a `fee` field.
```

File: _posts/en/newsletters/2021-01-06-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+  understand.  For example, a developer who wants to quickly implement
+  generic signmessage may choose to only implement signature verification for
+  P2PKH, P2WPKH, and P2SH-P2WPKH scripts, which covers a large
+  percentage of all current and historic outputs.  For anything else,
+  the wallet can return "inconclusive" and maybe direct the user to a
+  more comprehensive verification tool.
+
+- **Trampoline routing:** Bastien Teinturier [posted][teinturier post]
+  to the Lightning-Dev mailing list a new [specification][tramp spec]
+  for [trampoline routing][topic trampoline payments] which allows a
```
Comment (adamjonas) [inline _posts/en/newsletters/2021-01-06-newsletter.md:33]: nit: comma 👮 
```suggestion
  for [trampoline routing][topic trampoline payments], which allows a
```

File: _posts/en/newsletters/2021-01-06-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+## News
+
+- **Proposed updates to generic signmessage:** Andrew Poelstra
+  [posted][poelstra post] a partial rewrite of the [BIP322][]
+  generic message signing proposal.  The rewrite mainly clarifies the
+  signing and verification procedures, but it now also allows wallets
+  that don't implement a complete set of checks to return an
+  "inconclusive" state for signatures that use scripts they don't
+  understand.  For example, a developer who wants to quickly implement
+  generic signmessage may choose to only implement signature verification for
```
Comment (adamjonas) [inline _posts/en/newsletters/2021-01-06-newsletter.md:25]: nit: split infinitive

```suggestion
  understand.  For example, a developer who wants to implement generic
  signmessage quickly may choose to only implement signature verification for
```

**Review by adamjonas — COMMENTED** [review]
ACK modulo the update to the rc. Thanks, @harding & co!

File: _posts/en/newsletters/2021-01-06-newsletter.md (hunk 6 above)
Comment (jnewbery) [reply _posts/en/newsletters/2021-01-06-newsletter.md:127]: I don't think there's a problem with splitting infinitives. There's a good summary at https://en.wikipedia.org/wiki/Split_infinitive#Current_views.

For me, both this and the example below are clearer as they currently are - placing the modifier next to the verb that it's modifying.

**Comment by bitschmidty** [comment]
Squashed and merged. Great start to the year team. Really enjoying all the different contributors in addition to the new repo coverage!

File: _posts/en/newsletters/2021-01-06-newsletter.md (hunk 6 above)
Comment (adamjonas) [reply _posts/en/newsletters/2021-01-06-newsletter.md:129]: I don't think either suggestion diminishes the clarity, so I figured that these style suggestions were only a (marginal) net positive. However, this Wikipedia article has revealed that I'm an old-fashioned grammar luddite who needs to get with the times, so I'll stop flagging them.

**Pull request merged**